docs/REFERENCE.md

```diff
@@ -460,6 +460,10 @@ exist or is not a directory, then chezmoi will walk up the script's directory
 hierarchy and run the script in the first directory that exists and is a
 directory.
 
+**Scripts may have one prefix** - `after`, `before_`, and `once_`
+are exclusive. `before_once_` and `after_once_` are equivalent to
+`before_` or `after_`.
+
```

Comment:
This is not correct. All of the following a valid script prefixes:

```
run_
run_once_
run_once_before_
run_once_after_
run_before_
run_after_
```

`before_once_` is not the same as `before_`: `before_once_` is only run when the script's contents have changed, whereas `before_` scripts are run on every `chezmoi apply`.

The rest of the document does not use bold for emphasis.

Reply:
Oh! That changes my understanding entirely. I was using `run_(before|after)_once_` as the previous list indicates that they should come before `once_`. It seems to me that section would be worth updating, rather than this. I could split the prefixes into those that are valid for files and those for scripts and update the order. Does that align better?

Reply:
Yes, that's a great idea.

This line in the reference manual is wrong:

> The order of prefixes is important, the order is `run_`, `create_`, `modify_`, `before_`, `after_`, `exact_`, `private_`, `empty_`, `executable_`, `symlink_`, `once_`, `dot_`.